docs/contributing/building/macosx.md

```diff
@@ -1,98 +0,0 @@
-# Building Tracee in OSX
```

Comment:
Why removing the whole doc?
There are still parts here which are relevant for CO-RE as well (e.g. the recommendation to run vagrant)

Reply:
I'll get it back. Nobody seemed to be using/maintaining so I thought I could remove. No biggie if you prefer to keep, might be beneficial for someone playing with docker in OSX.

Reply:
OSX building instructions depend on kernel headers and non CORE object, as OSX Docker kernel does not have BTF embedded on it.

For WSL, for example, I have opened a bug upstream: https://github.com/microsoft/WSL/issues/9819 to enable kconfig options needed for Tracee (and eBPF progs).

> I believe those things should be fixed upstream so end user has a transparent experience with eBPF powered tools.

For Docker, there are some instructions to overcome docker kernel not having BTF info available. (Example: https://hemslo.io/run-ebpf-programs-in-docker-using-docker-bpf/).

So, **for OSX I would stick with Vagrant only**, with the file that we have in the repository (and an OSX user can use Vagrant to develop/collaborate with Tracee IMO).

Today morning I asked @geyslan to update the Vagrant file after this PR, as he is the biggest user from our team I believe (and capable of better maintaining the Vagrant file, maybe even giving instructions for OSX users in documentation ?).

Reply:
> Today morning I asked @geyslan to update the Vagrant file after this PR, as he is the biggest user from our team I believe (and capable of better maintaining the Vagrant file, maybe even giving instructions for OSX users in documentation ?).

👍🏼